lib/protocol/Writer.js

```diff
@@ -861,7 +861,7 @@ Writer.prototype[TypeCode.LONGDATE] = function writeLongDate(value) {
     hours = ~~ts[4];
     minutes = ~~ts[5];
     seconds = ~~ts[6];
-    nanoseconds = ~~((ts[6] * 1000000000) % 1000000000);
+    nanoseconds = ~~((ts[6] * 10000000) % 10000000);
```

Comment:
Alternatively, one could split `ts[6]` by the dot and convert the parts to a number or even adapt the timestamp regexp.